src/ipc/ipc4/handler.c

```diff
@@ -481,6 +483,22 @@ static int ipc4_set_pipeline_state(struct ipc4_message_request *ipc4)
 	return ret;
 }
 
+#if CONFIG_LIBRARY_MANAGER
+static int ipc4_load_library(struct ipc4_message_request *ipc4)
+{
+	struct ipc4_module_load_library library;
+	int ret = IPC4_UNAVAILABLE;
+
+	library.header.dat = ipc4->primary.dat;
+	library.data.dat = ipc4->extension.dat;
+
+	ret = lib_manager_load_library(library.header.r.dma_id, library.header.r.lib_id);
```

Comment:
The ret value is a type of POSIX (.e.g -EINVAL) not IPC4 like IPC4_INVALID_RESOURCE_ID. Please refine the error code in lib_load_manager to IPC4 ( is it only for IPC4 ?) or convert POSIX error code to IPC4 one

Reply:
This functionality is available only in closed source FW and firmware with IPC4. So that leads to why it is only IPC4. Code is split between FW and SoF module, that is why we have a mixture POSIX/IPC4. I've changed all ret from IPC4 to POSIX where I could, rest is not related to Zephyr.